open-craft/edx-platform
# [FC-0040] feat: update search index when object tags change

## Description

This PR implements updating search index with object tags whenever they change, based on the new `CONTENT_OBJECT_TAGS_CHANGED` event. 

## Supporting Information
- Related Ticket: https://github.com/openedx/modular-learning/issues/197
- Depends on: https://github.com/openedx/openedx-events/pull/327

## Testing Instructions
- Run your local devstack on this branch
- Clone the openedx-events repo on this branch https://github.com/openedx/openedx-events/pull/327 and make sure it is accessible from your devstack, and install it locally
  - `pip uninstall openedx-events`
  - `pip install -e path/to/openedx-events/`
- Make sure you have `meilisearch` setup locally, follow the setup instructions here https://github.com/open-craft/tutor-contrib-meilisearch
- Make sure you also have some sample taxonomies/tags, you can add some from here: https://github.com/open-craft/taxonomy-sample-data
- Run `tutor dev run cms bash` and `./manage.py cms reindex_studio`
- View the resulting index at http://meilisearch.local.edly.io:7700/ (see [tutor-contrib-meilisearch README](https://github.com/open-craft/tutor-contrib-meilisearch) for how to get the API key to log in)
- Check and confirm that is an index with documents in it. If you already have some tags applied to blocks, they will appear there
- Go back to studio, and add/update tags in a block
- Go back to meilisearch dashboard, and confirm that the `Tags` field in the index document for the block has been updated to reflect that changes you made.
  - Note: I recommend having a easily identifiable name for the block/unit that you added tags to so you can easily search for it
  - **NOTE:** Meilisearch seems to cache queries along with their results in the frontend, so if you simply search it might show you stale data (network tab shows no requests), especially if you're searching for the same query. Make sure you refresh the Meilisearch dashboard (http://meilisearch.local.edly.io:7700/) and then perform the search.
- Repeat and confirm the above for both units and blocks

---
Prinvate-ref: [FAL-3691](https://tasks.opencraft.com/browse/FAL-3691)
